src/content/blog/gpt2-mini-token-selection-visualization.md

```
@@ -0,0 +1,201 @@
+---
+title: "LLMの頭の中を覗く — GPT-2 miniで見る「次の文字予測」の仕組み"
+description: "自作GPT-2 miniの生成過程を可視化し、モデルがどのように確率分布を計算して次の文字を選んでいるか、Temperatureがどう影響するかを観測する"
+pubDate: 2026-01-09
+category: "engineering"
+---
+
+## LLM は何をしているのか
+
+ChatGPT や Claude などの LLM を使っていると、まるで「考えて」回答しているように見える。しかし、その本質は驚くほどシンプルだ。
+
+**次の文字（トークン）の確率分布を予測しているだけ**
```

markdownlint-cli2 (0.18.1)

Emphasis used instead of a heading

(MD036, no-emphasis-as-heading)